source/common/http/http1/codec_impl.cc

```diff
@@ -766,6 +771,7 @@ StatusOr<ParserStatus> ConnectionImpl::onHeadersComplete() {
       return codecProtocolError("http/1.1 protocol error: unsupported transfer encoding");
     }
   }
+  parser_->hasContentLength(request_or_response_headers.ContentLength() != nullptr);
```

Comment:
https://github.com/nodejs/llhttp/blob/aa07189d39b8d0da9d5a887d93761014a45804bf/test/request/content-length.md#error-on-simultaneous-content-length-and-transfer-encoding-identity

It's handled with an error. Actually this is just because llhttp doesn't have an initial value for unset content length. `http_parser`  uses all 1 bits set to indicate there is no content length. Added comment.

Reply:
We should have tests in envoy to cover these cases.  This is related to my question regarding use of lenient parsing further down in this file.  It seems that those lenient options were added for the benefit of Envoy's H1 implementation.

Reply:
https://github.com/envoyproxy/envoy/blob/56d3f8ab41cefbb6ca485f882f47b7e1703a3fa7/test/common/http/http1/codec_impl_test.cc#L1945 

There are tests for both content length and chunked.